lib/merchant_repository.rb

```diff
@@ -68,10 +68,14 @@ def invoices_by_merchant_id(id)
     @engine.invoices_by_merchant_id(id)
   end
 
+  def merchant_ids
+    @merchant_ids ||= all.map(&:id).sort
+  end
+
   private
 
   def max_merchant_id
-    all.max_by(&:id).id
+    @max_merchant_id ||= all.max_by(&:id).id
```

Comment:
I bet this helped speed up our code some. So we don't have to call max by over and over. Nice!